src/Bicep.LangServer.IntegrationTests/TelemetryTests.cs

```diff
@@ -339,18 +327,171 @@ public async Task BicepFileOpen_ShouldFireTelemetryEvent()
                     // location: resourceGroup().location
                     //   => Use a parameter here instead of 'resourceGroup().location'. 'resourceGroup().location' and 'deployment().location' should only be used as a default value for parameters.
                     "warnings",
-                    "2"
+                    "1"
                 },
                 { "modulesInReferencedFiles", "0" },
                 { "parentResourcesInReferencedFiles", "2" },
                 { "parametersInReferencedFiles", "2" },
                 { "variablesInReferencedFiles", "1" },
-                { "lineCountOfReferencedFiles", "12" }
+                { "lineCountOfReferencedFiles", "12" },
+                { "disableNextLineDirectivesCount", "4" },
+                { "diagnosticCodesInDisableNextLineDirectives", "{\"no-hardcoded-location\":2,\"BCP036\":1,\"BCP037\":1}" },
+                { "disableNextLineDirectivesCountInReferencedFiles", "0" },
+                { "diagnosticCodesInDisableNextLineDirectivesInReferencedFiles", "{}" }
+            };
+            await VerifyBicepFileOpenTelemetry(
+                bicepConfigFileContents,
+                mainBicepFileContents,
+                referencedBicepFileContents,
+                linterRuleStateOnBicepFileOpenTelemetryEventProperties,
+                bicepFileOpenTelemetryEventProperties);
+        }
+
+        [TestMethod]
+        public async Task BicepFileOpen_VerifyDisableNextLineDirectivesTelemetryPropertiesInReferencedFiles()
+        {
+            var mainBicepFileContents = @"param appInsightsName string = 'testAppInsightsName'
+
+var deployGroups = true
+
+resource applicationInsights 'Microsoft.Insights/components@2015-05-01' = {
+  name: appInsightsName
+#disable-next-line no-hardcoded-location
+  location: 'West US'
+  kind: 'web'
+  properties: {
+    Application_Type: 'web'
+  }
+  resource favorites 'favorites@2015-05-01'{
+    name: 'testName'
+  }
+}
+
+module apimGroups 'groups.bicep' = if (deployGroups) {
+  name: 'apimGroups'
+}";
+            var referencedBicepFileContents = @"resource parentAPIM 'Microsoft.ApiManagement/service@2019-01-01' existing = {
+  name: 'testApimInstanceName'
+}
+
+resource apimGroup 'Microsoft.ApiManagement/service/groups@2020-06-01-preview' = {
+  name: 'apiManagement/groups'
+}
+
+#disable-next-line no-unused-params
```

Comment:
It's possible to have multiple codes in the same directive, right?  Do you test that?

Reply:
Ah I missed that. Thanks!

Added a check and test as part of this commit:
https://github.com/Azure/bicep/pull/6151/commits/e851cc997271d236fcc23ce2f23963c1d40e71c8